torchdata/datapipes/iter/load/huggingface.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import posixpath
+
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple, Union
+
+from torch.utils.data.datapipes.utils.common import match_masks
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+from torchdata.datapipes.utils import StreamWrapper
+
+try:
+    import datasets
+except ImportError:
+    datasets = None
+
+def _get_response_from_huggingface_hub(dataset, split, revision, streaming, data_files) -> Tuple[Any, StreamWrapper]:
```

Comment:
The types of function arguments are missing, let's update them. 
Also, I'm curious to know the reason that we choose these 5 arguments, the original HF load_dataset API has a lot of [arguments](https://github.com/huggingface/datasets/blob/4844b9c65e042598e0bd9f2e5bd5b5f784c618bc/src/datasets/load.py#L1590-L1606)

Reply:
These seemed the most useful to me, I can add `**kwargs` so we can support all parameters

Reply:
Ok I don't think we need kwargs I added all parameters I could find listed on the main documentation https://huggingface.co/docs/datasets/loading

Reply:
There are 16. @ninginthecloud Do you think having those 5 should be sufficient?

https://huggingface.co/docs/datasets/package_reference/loading_methods#datasets.load_dataset